feat: improve token selector configuration and add buy on low balance feature OK-46990 OK-46991 OK-46992 OK-47426

Minnzen opened the pull request:
- Refactor token selector configuration: rename `perpTokenSortConfigPersistAtom` to `perpTokenSelectorConfigPersistAtom` for better clarity
- Update TokenSelector components to use new selector configuration with active tab state management
- Enhance sorting logic to maintain active tab state across components
- Add buy on low balance functionality in DepositWithdrawModal and SwapInputActions
- Integrate buy button to navigate to ReceiveModal when account balance is insufficient
- Add logging for buy actions to track user behavior

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Expanded token sort options: change % (24h), funding rate, volume (24h), open interest.
  * Added a contextual "Buy" flow in Deposit/Withdraw and Swap when balance is low.
  * New analytics/logging event for buy-on-low-balance actions.

* **Updates**
  * Token selector now persists active tab, sort field/direction and shares state across views.
  * Added sensible default tab and sort settings and preserved tab when toggling sorts.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Minnzen committed:
feat: implement buy on low balance functionality in DepositWithdrawModal and SwapInputActions

- Added a new logger method to track buy actions when the account balance is insufficient.
- Integrated buy button functionality in DepositWithdrawModal and SwapInputActions to navigate to the ReceiveModal with relevant token and account details.
- Enhanced user experience by providing actionable feedback for low balance scenarios.

Minnzen committed:
refactor: rename and update token sorting configuration in TokenSelector components

- Replaced `perpTokenSortConfigPersistAtom` with `perpTokenSelectorConfigPersistAtom` for clarity.
- Updated related components to use the new selector configuration, including state management for active tabs and sorting fields.
- Enhanced the sorting logic to maintain active tab state across the TokenSelector components.
- Adjusted type definitions to reflect the new configuration structure, ensuring consistency in the application.

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/2d3b595d-4f99-4bc1-9f40-0e5a138eaa74?refs=6f88aff9bf3b2a97b86cf11d0a5707b77579c85f&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/2d3b595d-4f99-4bc1-9f40-0e5a138eaa74/license?refs=6f88aff9bf3b2a97b86cf11d0a5707b77579c85f&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

originalix commented on packages/kit/src/views/Perp/components/TradingPanel/modals/DepositWithdrawModal.tsx:
```diff
@@ -388,6 +390,48 @@ function DepositWithdrawContent({
+      return;
+    }
+
+    void defaultLogger.wallet.walletActions.buyOnLowBalance({
```
日志不用刻意加 void

originalix commented on packages/kit/src/views/Perp/components/TradingPanel/modals/DepositWithdrawModal.tsx:
```diff
@@ -1332,6 +1376,28 @@ function DepositWithdrawContent({
+        isInsufficientBalance &&
+        selectedAction === 'deposit' &&
+        checkAccountSupport &&
+        !balanceLoading ? (
```
渲染的判断条件太多了，可以抽成 displayErrorMessage 的 useMemo 变量，阅读起来清晰一些

Minnzen committed:
Merge branch 'x' into feat/buy-token-update

Minnzen committed:
refactor: optimize buy button visibility logic in DepositWithdrawModal and SwapInputActions

- Simplified the condition for displaying the buy button in DepositWithdrawModal by using a memoized value.
- Updated the buyOnLowBalance function call to remove unnecessary async handling in both DepositWithdrawModal and SwapInputActions.

Minnzen committed:
refactor: introduce default values for token selector configuration

- Added constants for default values in the token selector, including active tab, sort field, and sort direction.
- Updated MobileTokenSelector and PerpTokenSelector components to utilize these new default values for improved consistency and maintainability.

Minnzen committed:
Merge branch 'x' into feat/buy-token-update
